atomic_reactor/plugins/post_group_manifests.py

```diff
@@ -0,0 +1,212 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+
+get the image manifest lists from the worker builders. If possible, group them together
+and return them. if not, return the x86_64/amd64 image manifest instead after re-uploading
+it for all existing image tags.
+"""
+
+
+from __future__ import unicode_literals
+import requests
+import requests.auth
+from tempfile import NamedTemporaryFile
+import yaml
+import subprocess
+
+from six.moves.urllib.parse import urlparse
+
+from atomic_reactor.plugin import PostBuildPlugin, PluginFailedException
+from atomic_reactor.util import Dockercfg, get_manifest_digests
+
+
+class GroupManifestsPlugin(PostBuildPlugin):
+    key = 'group_manifests'
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, registries, group=True, goarch=None):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param registries: dict, keys are docker registries, values are dicts containing
+                           per-registry parameters.
+                           Params:
+                            * "secret" optional string - path to the secret, which stores
+                              login and password for remote registry
+        :param group: bool, if true, create a manifest list; otherwise only add tags to
+                      amd64 image manifest
+        :param goarch: dict, keys are platform, values are go language platform names
+        """
+        # call parent constructor
+        super(GroupManifestsPlugin, self).__init__(tasker, workflow)
+        self.group = True # group
+        self.goarch = goarch or {}
+        self.registries = registries
+        self.worker_registries = {}
+
+    def submit_manifest_list(self, registry, registry_conf, manifest_list_spec):
+        docker_secret_path = registry_conf.get('secret', None)
+#        dockercfg = "{0}/.dockercfg".format(registry_conf.get('secret', None))
+        
+        with NamedTemporaryFile(prefix='manifest-list', suffix=".yml", mode='w') as fp:
+            yaml.dump(manifest_list_spec, stream=fp)
+            fp.flush()
+            self.log.debug("Wrote to file %s with config %s", fp.name, docker_secret_path)
+
+            cmd = ['manifest-tool', '--docker-cfg=%s' % docker_secret_path,
```

Comment:
Do we think `--docker-cfg` is useful here at all, given how docker/cli behaves with .dockercfg?

It seems to me that only the newer-style 'config.json' file is read from the --docker-cfg location.